runner/sidecar/sources.go

```diff
@@ -45,6 +45,11 @@ func connectSources(ctx context.Context, toMain func(context.Context, []byte) er
 		if leadReplica() { // only replica zero updates this value, so it the only replica that can be accurate
 			newSourceMetrics(source, sourceName)
 		}
+		retryCountMetrics := promauto.NewCounter(prometheus.CounterOpts{
+			Subsystem: "message",
+			Name:      "retry-counts",
```

Comment:
invalid name - must be underscore

Reply:
I think this should move into `newSourceMetrics`. To do this, we need to store the value in the step status, much like we do for errors. You'll need to change `metrics.go` to add this.